src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/LinkedCasesMapping.java

```diff
@@ -20,17 +23,66 @@ private LinkedCasesMapping() {
 
     public static List<ServiceLinkedCases> getLinkedCases(SscsCaseData caseData) {
         return Optional.ofNullable(caseData.getLinkedCase())
-                .orElseGet(Collections::emptyList).stream()
-                .filter(Objects::nonNull)
-                .map(CaseLink::getValue)
-                .filter(Objects::nonNull)
-                .map(CaseLinkDetails::getCaseReference)
-                .filter(StringUtils::isNotBlank)
-                .map(caseReference -> ServiceLinkedCases.builder()
-                    .caseReference(caseReference)
-                    .caseName(HearingsCaseMapping.getPublicCaseName(caseData))
-                    .reasonsForLink(HearingsCaseMapping.getReasonsForLink(caseData))
-                    .build())
-                .collect(Collectors.toList());
+            .orElseGet(Collections::emptyList).stream()
```

Comment:
I think having this here will work but you should be able to move the or else to the bottom. I believe ofNullable will falldown to the orElse once it hits a null value. Same as below

Reply:
Discussed on a call. Happy to close